webviz_subsurface/_datainput/eclipse_init_io/pvt_common.py

```diff
@@ -0,0 +1,872 @@
+    """
+    A namespace for constant definitions for
+    reading Eclipse INIT files.
+    """
+
+    LOGIHEAD_KW = "LOGIHEAD"
+    INTEHEAD_KW = "INTEHEAD"
+    INTEHEAD_UNIT_INDEX = 2
+    INTEHEAD_PHASE_INDEX = 14
+    LOGIHEAD_RS_INDEX = 0
+    LOGIHEAD_RV_INDEX = 1
+    TABDIMS_IBPVTO_OFFSET_ITEM = 6
+    TABDIMS_JBPVTO_OFFSET_ITEM = 7
+    TABDIMS_NRPVTO_ITEM = 8
+    TABDIMS_NPPVTO_ITEM = 9
+    TABDIMS_NTPVTO_ITEM = 10
+    TABDIMS_IBPVTW_OFFSET_ITEM = 11
+    TABDIMS_NTPVTW_ITEM = 12
+    TABDIMS_IBPVTG_OFFSET_ITEM = 13
+    TABDIMS_JBPVTG_OFFSET_ITEM = 14
+    TABDIMS_NRPVTG_ITEM = 15
+    TABDIMS_NPPVTG_ITEM = 16
+    TABDIMS_NTPVTG_ITEM = 17
+    LOGIHEAD_CONSTANT_OILCOMPR_INDEX = 39 - 1
+
+    TABDIMS_IBDENS_OFFSET_ITEM = 18
+    TABDIMS_NTDENS_ITEM = 19
+
+
+class EclPhaseIndex(Enum):
+    """Enumerator holding the different phases according
+    to Eclipse file conventions"""
+
+    Aqua = 0
+    Liquid = 1
+    Vapour = 2
+
+
+def is_const_compr_index() -> int:
+    """Convenient function for better readibility.
+
+    Returns:
+        An integer that states that the oil has constant compression
+        according to Eclipse LOGIHEAD file conventions.
+    """
+    return InitFileDefinitions.LOGIHEAD_CONSTANT_OILCOMPR_INDEX
+
+
+def surface_mass_density(ecl_file: EclFile, phase: EclPhaseIndex) -> List[float]:
+    """Extracts the surface mass density from the given Eclipse file for the given phase.
+
+    Args:
+        ecl_file: The Eclipse file to extract data from
+        phase: Fluid phase to extract data for
+
+    Returns:
+        List of surface mass density values.
+
+    """
+    if phase is EclPhaseIndex.Liquid:
+        col = 0
+    elif phase is EclPhaseIndex.Aqua:
+        col = 1
+    elif phase is EclPhaseIndex.Vapour:
+        col = 2
+    else:
+        raise AttributeError("Phase must be Liquid, Aqua or Vapour.")
+
+    tabdims = ecl_file.__getitem__("TABDIMS")
+    tab = ecl_file.__getitem__("TAB")
+
+    start = tabdims[InitFileDefinitions.TABDIMS_IBDENS_OFFSET_ITEM] - 1
+    nreg = tabdims[InitFileDefinitions.TABDIMS_NTDENS_ITEM]
+
+    rho = [tab[start + nreg * (col + 0)], tab[start + nreg * (col + 1)]]
+
+    return rho
+
+
+class FluidImplementation:
+    """Base class for fluid implementations
+
+    Holds a list of regions (one per PVT table).
+
+    Attributes:
+        keep_unit_system: True if the original unit system was kept
+        original_unit_system: An ErtEclUnitEnum representing the original unit system
+    """
+
+    class InvalidArgument(Exception):
+        """An exception for invalid arguments"""
+
+        def __init__(self, message: str):
+            self.message = message
+            super().__init__(message)
+
+    class InvalidType(Exception):
+        """An exception for invalid types"""
+
+        def __init__(self) -> None:
+            super().__init__("Invalid type. Only live oil/wet gas/water supported.")
+
+    def __init__(self, keep_unit_system: bool = False) -> None:
```

Comment:
Applies several places, but this is the base class. Don't you think we by default should keep the unit system from the input?
```suggestion
    def __init__(self, keep_unit_system: bool = True) -> None:
```